multiqc/modules/rockhopper/rockhopper.py

```diff
@@ -76,30 +89,44 @@ def parse_rockhopper_summary(self,f):
         ]
 
         results = { name:0 for name in stats_index }
+        # Files can have more than one sample in them. Store results for
+        # each sample in a dictionary
+        results_by_s_name = {}
 
         # Parse rockhopper output line-by-line since there may be many genomes
         lines = f['f'].split('\n')
         for i, line in enumerate(lines):
 
             # Get sample name
             if line.startswith('Aligning sequencing reads from file:'):
+                # When reaching a new sample add the previous sample to
+                # the dictionary of samples along with its results.
+                # Then reset the results dictionary for the new sample
+                if s_name  and s_name not in results_by_s_name:
+                    results_by_s_name[s_name] = results
+                    results = { name:0 for name in stats_index }
+
                 s_name = line.split(':', 1)[1].strip()
                 s_name = self.clean_s_name(s_name,f['root'])
 
             elif line.startswith('Aligning sequencing reads from files:'):
+                if s_name and s_name not in results_by_s_name:
+                    results_by_s_name[s_name] = results
+                    results = { name:0 for name in stats_index }
                 s_name = lines[i+1].strip()
                 s_name = self.clean_s_name(s_name,f['root'])
 
             # Get total number of reads read by rockhopper
             if line.startswith('Total reads:'):
                 results['total-reads'] = int(re.search('Total reads:\s*(\d*)',line).group(1))
-                i += 1
 
             # Get number of reads aligned to each genome
             elif line.startswith('Successfully aligned reads'):
 
                 # Get Genome ID
-                genome = re.search('\(>(.*?) .*\)',line).group(1)
+                # This isn't used for anything but could be used to make
```

Comment:
If it's not used then let's get rid of it.

Reply:
Agreed. Thanks!